store/src/java/com/zimbra/cs/service/FileUploadServlet.java

```diff
@@ -881,10 +881,52 @@ private static long getFileUploadMaxSize(boolean limitByFileUploadMaxSize) {
         return maxSize;
     }
 
+    private static long getFileUploadMaxSize(boolean limitByFileUploadMaxSize, Account acct) {
+        long maxSize = DEFAULT_MAX_SIZE;
+        String attributeUsed = null;
+        try {
+            if (limitByFileUploadMaxSize && acct.getFileUploadMaxSizePerFile() > 0) {
+                long fileUploadSizePerFile = acct.getFileUploadMaxSizePerFile();
+                long fileUploadSize = Provisioning.getInstance().getLocalServer().getLongAttr(
+                        Provisioning.A_zimbraFileUploadMaxSize, DEFAULT_MAX_SIZE);
+                if (fileUploadSizePerFile >= fileUploadSize) {
+                    maxSize = fileUploadSize;
+                    attributeUsed = Provisioning.A_zimbraFileUploadMaxSize;
+                } else {
+                    maxSize = fileUploadSizePerFile;
+                    attributeUsed = Provisioning.A_zimbraFileUploadMaxSizePerFile;
+                }
+            } else if (!limitByFileUploadMaxSize && acct.getMailAttachmentMaxSize() > 0) {
+                long mailAttachmentMaxSize = acct.getMailAttachmentMaxSize();
+                long mtaMaxMsgSize = Provisioning.getInstance().getConfig().getLongAttr(
+                        Provisioning.A_zimbraMtaMaxMessageSize, DEFAULT_MAX_SIZE);
+                if (mailAttachmentMaxSize > mtaMaxMsgSize) {
+                    maxSize = mtaMaxMsgSize;
+                    attributeUsed = Provisioning.A_zimbraMtaMaxMessageSize;
+                } else {
+                    maxSize = mailAttachmentMaxSize;
+                    attributeUsed = Provisioning.A_zimbraMailAttachmentMaxSize;
+                }
+                if (maxSize == 0) {
```

Comment:
In what condition maxSize can be zero, and what does -1 means, one liner comment please.

Reply:
`0` means `no limit`.  The return value from this function gets used by `FileUploadBase` `sizeMax` where `-1` means `no limit`